sky/cli.py

```diff
@@ -1675,6 +1676,18 @@ def _get_spot_jobs(
                     'clusters, the returned IP address is the internal IP '
                     'of the head pod, and may not be accessible from outside '
                     'the cluster.'))
+@click.option('--endpoints',
+              default=False,
+              is_flag=True,
+              required=False,
+              help=('Get all exposed endpoints and corresponding URLs for a'
+                    'cluster. This option will override all other options.'))
+@click.option('--endpoint',
```

Comment:
I ran into a bug when `--endpoint/s` is called on a k8s cluster without any ports open. Repro:

1. `sky launch -c dbg --cloud kubernetes`

2. `sky status --endpoint 8888 dbg`
```
(base) ➜  ~ sky status --endpoint 8888 dbg
Traceback (most recent call last):
  File "/Users/romilb/tools/anaconda3/bin/sky", line 8, in <module>
    sys.exit(cli())
  File "/Users/romilb/tools/anaconda3/lib/python3.9/site-packages/click/core.py", line 1128, in __call__
    return self.main(*args, **kwargs)
  File "/Users/romilb/tools/anaconda3/lib/python3.9/site-packages/click/core.py", line 1053, in main
    rv = self.invoke(ctx)
  File "/Users/romilb/Romil/Berkeley/Research/sky-experiments/sky/utils/common_utils.py", line 315, in _record
    return f(*args, **kwargs)
  File "/Users/romilb/Romil/Berkeley/Research/sky-experiments/sky/cli.py", line 1168, in invoke
    return super().invoke(ctx)
  File "/Users/romilb/tools/anaconda3/lib/python3.9/site-packages/click/core.py", line 1659, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/romilb/tools/anaconda3/lib/python3.9/site-packages/click/core.py", line 1395, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/romilb/tools/anaconda3/lib/python3.9/site-packages/click/core.py", line 754, in invoke
    return __callback(*args, **kwargs)
  File "/Users/romilb/Romil/Berkeley/Research/sky-experiments/sky/utils/common_utils.py", line 336, in _record
    return f(*args, **kwargs)
  File "/Users/romilb/Romil/Berkeley/Research/sky-experiments/sky/cli.py", line 1951, in status
    port_details = provision_lib.query_ports(
  File "/Users/romilb/Romil/Berkeley/Research/sky-experiments/sky/provision/__init__.py", line 41, in _wrapper
    return impl(*args, **kwargs)
  File "/Users/romilb/Romil/Berkeley/Research/sky-experiments/sky/provision/kubernetes/network.py", line 167, in query_ports
    return _query_ports_for_loadbalancer(
  File "/Users/romilb/Romil/Berkeley/Research/sky-experiments/sky/provision/kubernetes/network.py", line 189, in _query_ports_for_loadbalancer
    external_ip = network_utils.get_loadbalancer_ip(
  File "/Users/romilb/Romil/Berkeley/Research/sky-experiments/sky/provision/kubernetes/network_utils.py", line 176, in get_loadbalancer_ip
    service = core_api.read_namespaced_service(
  File "/Users/romilb/tools/anaconda3/lib/python3.9/site-packages/kubernetes/client/api/core_v1_api.py", line 25141, in read_namespaced_service
    return self.read_namespaced_service_with_http_info(name, namespace, **kwargs)  # noqa: E501
  File "/Users/romilb/tools/anaconda3/lib/python3.9/site-packages/kubernetes/client/api/core_v1_api.py", line 25228, in read_namespaced_service_with_http_info
    return self.api_client.call_api(
  File "/Users/romilb/tools/anaconda3/lib/python3.9/site-packages/kubernetes/client/api_client.py", line 348, in call_api
    return self.__call_api(resource_path, method,
  File "/Users/romilb/tools/anaconda3/lib/python3.9/site-packages/kubernetes/client/api_client.py", line 180, in __call_api
    response_data = self.request(
  File "/Users/romilb/tools/anaconda3/lib/python3.9/site-packages/kubernetes/client/api_client.py", line 373, in request
    return self.rest_client.GET(url,
  File "/Users/romilb/tools/anaconda3/lib/python3.9/site-packages/kubernetes/client/rest.py", line 244, in GET
    return self.request("GET", url,
  File "/Users/romilb/tools/anaconda3/lib/python3.9/site-packages/kubernetes/client/rest.py", line 238, in request
    raise ApiException(http_resp=r)
kubernetes.client.exceptions.ApiException: (404)
Reason: Not Found
HTTP response headers: HTTPHeaderDict({'Audit-Id': '91e9d358-a57e-4e99-8e48-3d12d5a134e6', 'Cache-Control': 'no-cache, private', 'Content-Type': 'application/json', 'X-Kubernetes-Pf-Flowschema-Uid': '496771b9-87ba-44f7-ac29-a04db493ac38', 'X-Kubernetes-Pf-Prioritylevel-Uid': '803aa7b0-3b91-4fd5-9732-ca8136d4456f', 'Date': 'Thu, 16 Nov 2023 12:40:16 GMT', 'Content-Length': '236'})
HTTP response body: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"services \"dbg-2ea4-skypilot-loadbalancer\" not found","reason":"NotFound","details":{"name":"dbg-2ea4-skypilot-loadbalancer","kind":"services"},"code":404}
```

Reply:
Fixed in https://github.com/skypilot-org/skypilot/pull/2713/commits/db9fb6b96e5047e34fee26426c6d77f4eaf33ed4